ta/pkcs11/src/pkcs11_token.c

```diff
@@ -111,3 +111,54 @@ uint32_t entry_ck_slot_list(uint32_t ptypes, TEE_Param *params)
 
 	return PKCS11_CKR_OK;
 }
+
+uint32_t entry_ck_slot_info(uint32_t ptypes, TEE_Param *params)
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INOUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_MEMREF_OUTPUT,
+						TEE_PARAM_TYPE_NONE);
+	TEE_Param *ctrl = &params[0];
+	TEE_Param *out = &params[2];
+	uint32_t rv = 0;
+	struct serialargs ctrlargs = { };
+	uint32_t token_id = 0;
+	struct ck_token *token = NULL;
+	const char desc[] = PKCS11_SLOT_DESCRIPTION;
+	const char manuf[] = PKCS11_SLOT_MANUFACTURER;
+	const char hwver[2] = PKCS11_SLOT_HW_VERSION;
+	const char fwver[2] = PKCS11_SLOT_FW_VERSION;
+	struct pkcs11_slot_info info = { };
```

Comment:
I would have preferred initializing as many fields as possible of the struct here. That would save us from the `const char` variables above and the need of an extra `strcpy()` below.

Reply:
ok, sounds fair.